android/src/main/java/com/bluechilli/flutteruploader/UploadWorker.java

```diff
@@ -159,8 +159,11 @@ public Result doWorkInternal() {
                   null));
         }
 
-        String mimeType = GetMimeType(item.getPath());
-        MediaType contentType = MediaType.parse(mimeType);
+        MediaType contentType = null;
+        if (headers == null || (headers != null && headers.get("Content-Type") == null && headers.get("content-type") == null)) {
```

Comment:
The second check for `headers != null` is not needed.

Why the check for "content-type"? The http standard defines `Content-Type`, correct?

Reply:
https://stackoverflow.com/a/5259004
It's case insensitive.
So to be completely right need to check for all possible keys.

Reply:
OK thank you!
In that case it may be "safer" to extract all keys, convert to lowercase and merge a lowercased input map with it. 
Currently, the suggested change will still have issues with `Content-type`, `ConTent-TYPE` etc.